docs/ldml/tr35-keyboards.md

```diff
@@ -308,6 +308,263 @@ For extensibility, the `<special>` element will be allowed at nearly every level
+        1. break out of the loop
+    2. If _p_.divider? == false:
+        1. Prepend marker _p_.marker to the output string _o_
+    3. Set _p_.processed?=true (so we don't process this again)
+2. Loop through each codepoint _i_ ( in the plain text input string ) from end to beginning (backwards)
+    1. Prepend _i_ to output _o_
+    2. Loop through the elements _p_ of the array _e_ from end to beginning (backwards)
+        1. If _p_.processed? == true:
+            1. Continue the inner loop  (was already processed)
+        2. If _p_.glue isn't _i_
+            1. Continue the inner loop  (wrong glue, not applicable)
+        3. If _p_.divider? == true:
+            1. Break out of the inner loop  (reached end of this 'glue' char)
+        4. Prepend marer _p_.marker to the output string _o_
+        5. Set _p_.processed?=true (so we don't process this again)
+3. _o_ is now the output string including markers.
+
+#### Example Normalization with Markers
+
+**Example 1**
+
+Consider this example, without markers:
+
+- `e\u{0300}\u{0320}` (input)
+- `e\u{0320}\u{0300}` (NFD)
+
+The combining marks are reordered.
+
+**Example 2**
+
+If we add markers:
+
+- `e\u{0300}\m{marker}\u{0320}` (input)
+- `e\m{marker}\u{0320}\u{0300}` (NFD)
+
+Note that the marker is 'glued' to the _following_ character. In the above example, `\m{marker}` was 'glued' to the `\u{0320}`.
+
+**Example 2**
+
+A second example:
+
+- `e\m{marker0}\u{0300}\m{marker1}\u{0320}\m{marker2}` (input)
+- `e\m{marker1}\u{0320}\m{marker0}\u{0300}\m{marker2}` (NFD)
+
+Here `\m{marker2}` is 'glued' to the end of the string. However, if additional text is added such as by a subsequent keystroke (which may add an additional combining character, for example), this marker may be 'glued' to that following text.
+
+Markers remain in the same normalization-safe segment during normalization. Consider:
+
+**Example 3**
+
+- `e\u{0300}\m{marker1}\u{0320}a\u{0300}\m{marker2}\u{0320}` (original)
+- `e\m{marker1}\u{0320}\u{0300}a\m{marker2}\u{0320}\u{0300}` (NFD)
+
+There are two normalization-safe segments here:
+
+1. `e\u{0300}\m{marker1}\u{0320}`
+2. `a\u{0300}\m{marker2}\u{0320}`
+
+Normalization (and marker rearranging) effectively occurs within each segment.  While `\m{marker1}` is 'glued' to the `\u{0320}`, it is glued within the first segment and has no effect on the second segment.
+
+### Normalization and Character Classes
+
+If pre-composed (non-NFD) characters are used in [character classes](#regex-like-syntax), such as `[á-é]`, these may not match as keyboard authors expect, as the U+00E1 character will not occur in NFD form. Thus this may be masking serious errors in the data.
+
+By default, tools that process the keyboard data should raise an error when character classes include non-NFD characters.
+
+The above should be written instead as a regex `(á|â|ã|ä|å|æ|ç|è|é)`, or as a set variable `<set id="Example" value="á â ã ä å æ ç è é"/>` and matched as `$[Example]`.
```

Comment:
```suggestion
The above should be written instead as a regex `(á|â|ã|ä|å|æ|ç|è|é)`. Alternative it could be written as a set variable `<set id="Example" value="á â ã ä å æ ç è é"/>` and matched as `$[Example]`.
```

Slightly confusing unless we split the solutions.

Reply:
```suggestion
The above should be written instead as a regex `(á|â|ã|ä|å|æ|ç|è|é)`. Alternatively, it could be written as a set variable `<set id="Example" value="á â ã ä å æ ç è é"/>` and matched as `$[Example]`.
```

Reply:
thx for fixing my fix @jahorton 🤣

Reply:
thanks!